rules/prefer-number-is-integer.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+const isSameReference = require('./utils/is-same-reference.js');
+
+const MESSAGE_ID = 'preferNumberIsInteger';
+const MESSAGE_ID_SUGGEST = 'preferNumberIsIntegerSuggestion';
+const messages = {
+	[MESSAGE_ID]: 'Replace `{{original}}` with `Number.isInteger({{variable}})`.',
+	[MESSAGE_ID_SUGGEST]: 'Prefer `Number.isInteger()` for integer checks.',
+};
+
+const equalsSelector = ':matches([operator="==="],[operator="=="])';
+
+// Value % 1 === 0
+const modulo1Selector = [
+	'BinaryExpression',
+	'[left.type="BinaryExpression"]',
+	'[left.operator="%"]',
+	'[left.right.value=1]',
+	equalsSelector,
+	'[right.value="0"]',
+].join('');
+
+// (value^0) === value OR (value | 0) === value
+const mathOperatorSelector = [
+	'BinaryExpression',
+	'[left.type="BinaryExpression"]',
+	`:matches(${['^', '|'].map(operator => `[left.operator="${operator}"]`).join(',')})`,
+	'[left.right.value=0]',
+	equalsSelector,
+].join('');
+
+// ParseInt(value,10) === value
```

Comment:
```suggestion
// `Number.parseInt(value, 10) === value`
```

Let's forget about `parseInt`, only check `Number.parseInt`